man/man5/zfs-module-parameters.5

```diff
@@ -2995,6 +2995,18 @@ must be at least twice the maximum block size in use.
 Default value: \fB16,777,216\fR.
 .RE
 
+.sp
+.ne 2
+.na
+\fBzfs_recv_write_batch_size\fR (int)
```

Comment:
This is lsize, not psize, right? Should we specify that?  Would there be an advantage to doing psize instead, since it correlates to the actual amount of data being written to disk?

Reply:
This is file-offset bytes, which is most similar to lsize in that it is not compressed.  It would be difficult to do this based on compressed size, because the DMU does a bunch of accounting based on file-bytes (e.g. DMU_MAX_ACCESS).  I can update this manpage to be more explicit (but TBH, figuring out how to modify most of the module parameters requires reading the source code).

Reply:
> I can update this manpage to be more explicit (but TBH, figuring out how to modify most of the module parameters requires reading the source code).

IMO that's a bug, not a feature; I think if we want ZFS to be widely accessible we should work on making these man pages more usable to non-devs.

Reply:
I updated the comment to be a bit more descriptive:

> The maximum amount of data (in bytes) that \fBzfs receive\fR will write in
one DMU transaction.  This is the uncompressed size, even when receiving a
compressed send stream.  This setting will not reduce the write size below
a single block. Capped at a maximum of 32MB